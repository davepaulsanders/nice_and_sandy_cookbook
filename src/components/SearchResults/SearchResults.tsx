import RecipeCard from "../RecipeCard/RecipeCard"
import Wrapper from "../Wrapper/Wrapper"

const SearchResults = ({recipeSearchCopy, setRecipeSearchCopy, setRecipes }) => {
	return (
		<Wrapper>
			{recipeSearchCopy.map((recipe) =>  <RecipeCard {...recipe} setRecipeSearchCopy={setRecipeSearchCopy} setRecipes={setRecipes}/>)}
		</Wrapper>
		   )
}
export default SearchResults
