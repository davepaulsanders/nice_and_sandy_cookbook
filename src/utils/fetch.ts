export const fetchData = async (url: string) => {
	const resp = await fetch(url)
	if (resp.status !== 200) throw new Error(`Request failed with ${resp.status}`)
	const data = resp.json()
	return data
}
