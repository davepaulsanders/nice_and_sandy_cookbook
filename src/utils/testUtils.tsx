import RecipeCard from '../components/RecipeCard/RecipeCard';
import { ReactElement } from "react"
import { vi } from "vitest"
import { Recipe } from "../types/types"

const generateRecipes = () => {
const recipes: Recipe[] = []
for (let i = 0; i < 10; i++) {
	recipes.push({
	id:i + 1,
	img:"https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg",
	href:"https://www.feastingathome.com/vegan-fajitas/",
	label:`Vegan Fajitas ${i + 1}`,
	alt:"test",
	category:"ENTREE",
	category_id:1,
	is_pinned: false,
	})
}
return recipes
}
const generateRecipeCards = () => {
const recipeCards: ReactElement[] = []
for (let i = 0; i < 10; i++) {
	recipeCards.push(
	<RecipeCard
	id={i + 1}
	img="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
	href="https://www.feastingathome.com/vegan-fajitas/"
	label="Vegan Fajitas 2"
	alt="test"
	category="ENTREE"
	category_id={1}
	is_pinned= {false}
	recipes={generateRecipes()}
	setRecipes={vi.fn()}
	/>)
	}
	return recipeCards
}
export { generateRecipeCards, generateRecipes }
