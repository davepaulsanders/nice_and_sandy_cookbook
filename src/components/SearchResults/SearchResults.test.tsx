import SearchResults from "./SearchResults";
import { expect, test, vi } from "vitest";
import { render } from "@testing-library/react";

import { generateRecipes } from "../../utils/testUtils";

const recipes = generateRecipes();

import "@testing-library/dom";

global.fetch = vi.fn((url: string) => {
	if (url === "../v1/recipes") {
		return Promise.resolve({
			status: 200,
			json: () => Promise.resolve({ recipes: generateRecipes() }),
		});
	}
	if (url === "../v1/categories") {
		return Promise.resolve({
			status: 200,
			json: () => Promise.resolve({ categories: ["ENTREE"] }),
		});
	}
	return Promise.resolve({
		json: () => Promise.resolve({ error: "Unknown path" }),
	});
}) as unknown as typeof global.fetch;

test("<SearchResults /> renders", () => {
	const { container } = render(
		<SearchResults
			recipeSearchCopy={recipes}
			setRecipeSearchCopy={vi.fn()}
			setRecipes={vi.fn()}
		/>
	);
	expect(container).toMatchSnapshot();
});
