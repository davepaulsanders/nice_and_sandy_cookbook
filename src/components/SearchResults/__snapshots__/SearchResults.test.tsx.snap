// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SearchResults /> renders 1`] = `
<div>
  <div
    class="grid gap-4 grid-cols-2 sm:grid-cols-6 sm:mx-auto"
  >
    <a
      href="https://www.feastingathome.com/vegan-fajitas/"
      target="blank"
    >
      <div
        class="cursor-pointer relative flex flex-col justify-content-center \\   items-center "
      >
        <div
          class="mt-4 w-full rounded-md overflow-hidden"
        >
          <img
            alt="test"
            class="h-[16rem] sm:h-[20rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-2"
        >
          <p
            class="text-lightg text-md"
          >
            Vegan Fajitas 1
          </p>
        </div>
        <img
          class="absolute top-5 right-1 hover:scale-125 hover:cursor-pointer"
          data-testid="pin"
          height="30px"
          src="/thumb-tack.webp"
          width="30px"
        />
      </div>
    </a>
    <a
      href="https://www.feastingathome.com/vegan-fajitas/"
      target="blank"
    >
      <div
        class="cursor-pointer relative flex flex-col justify-content-center \\   items-center "
      >
        <div
          class="mt-4 w-full rounded-md overflow-hidden"
        >
          <img
            alt="test"
            class="h-[16rem] sm:h-[20rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-2"
        >
          <p
            class="text-lightg text-md"
          >
            Vegan Fajitas 2
          </p>
        </div>
        <img
          class="absolute top-5 right-1 hover:scale-125 hover:cursor-pointer"
          data-testid="pin"
          height="30px"
          src="/thumb-tack-cancel.webp"
          width="30px"
        />
      </div>
    </a>
    <a
      href="https://www.feastingathome.com/vegan-fajitas/"
      target="blank"
    >
      <div
        class="cursor-pointer relative flex flex-col justify-content-center \\   items-center "
      >
        <div
          class="mt-4 w-full rounded-md overflow-hidden"
        >
          <img
            alt="test"
            class="h-[16rem] sm:h-[20rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-2"
        >
          <p
            class="text-lightg text-md"
          >
            Vegan Fajitas 3
          </p>
        </div>
        <img
          class="absolute top-5 right-1 hover:scale-125 hover:cursor-pointer"
          data-testid="pin"
          height="30px"
          src="/thumb-tack-cancel.webp"
          width="30px"
        />
      </div>
    </a>
    <a
      href="https://www.feastingathome.com/vegan-fajitas/"
      target="blank"
    >
      <div
        class="cursor-pointer relative flex flex-col justify-content-center \\   items-center "
      >
        <div
          class="mt-4 w-full rounded-md overflow-hidden"
        >
          <img
            alt="test"
            class="h-[16rem] sm:h-[20rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-2"
        >
          <p
            class="text-lightg text-md"
          >
            Vegan Fajitas 4
          </p>
        </div>
        <img
          class="absolute top-5 right-1 hover:scale-125 hover:cursor-pointer"
          data-testid="pin"
          height="30px"
          src="/thumb-tack-cancel.webp"
          width="30px"
        />
      </div>
    </a>
    <a
      href="https://www.feastingathome.com/vegan-fajitas/"
      target="blank"
    >
      <div
        class="cursor-pointer relative flex flex-col justify-content-center \\   items-center "
      >
        <div
          class="mt-4 w-full rounded-md overflow-hidden"
        >
          <img
            alt="test"
            class="h-[16rem] sm:h-[20rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-2"
        >
          <p
            class="text-lightg text-md"
          >
            Vegan Fajitas 5
          </p>
        </div>
        <img
          class="absolute top-5 right-1 hover:scale-125 hover:cursor-pointer"
          data-testid="pin"
          height="30px"
          src="/thumb-tack-cancel.webp"
          width="30px"
        />
      </div>
    </a>
    <a
      href="https://www.feastingathome.com/vegan-fajitas/"
      target="blank"
    >
      <div
        class="cursor-pointer relative flex flex-col justify-content-center \\   items-center "
      >
        <div
          class="mt-4 w-full rounded-md overflow-hidden"
        >
          <img
            alt="test"
            class="h-[16rem] sm:h-[20rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-2"
        >
          <p
            class="text-lightg text-md"
          >
            Vegan Fajitas 6
          </p>
        </div>
        <img
          class="absolute top-5 right-1 hover:scale-125 hover:cursor-pointer"
          data-testid="pin"
          height="30px"
          src="/thumb-tack-cancel.webp"
          width="30px"
        />
      </div>
    </a>
    <a
      href="https://www.feastingathome.com/vegan-fajitas/"
      target="blank"
    >
      <div
        class="cursor-pointer relative flex flex-col justify-content-center \\   items-center "
      >
        <div
          class="mt-4 w-full rounded-md overflow-hidden"
        >
          <img
            alt="test"
            class="h-[16rem] sm:h-[20rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-2"
        >
          <p
            class="text-lightg text-md"
          >
            Vegan Fajitas 7
          </p>
        </div>
        <img
          class="absolute top-5 right-1 hover:scale-125 hover:cursor-pointer"
          data-testid="pin"
          height="30px"
          src="/thumb-tack-cancel.webp"
          width="30px"
        />
      </div>
    </a>
    <a
      href="https://www.feastingathome.com/vegan-fajitas/"
      target="blank"
    >
      <div
        class="cursor-pointer relative flex flex-col justify-content-center \\   items-center "
      >
        <div
          class="mt-4 w-full rounded-md overflow-hidden"
        >
          <img
            alt="test"
            class="h-[16rem] sm:h-[20rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-2"
        >
          <p
            class="text-lightg text-md"
          >
            Vegan Fajitas 8
          </p>
        </div>
        <img
          class="absolute top-5 right-1 hover:scale-125 hover:cursor-pointer"
          data-testid="pin"
          height="30px"
          src="/thumb-tack-cancel.webp"
          width="30px"
        />
      </div>
    </a>
    <a
      href="https://www.feastingathome.com/vegan-fajitas/"
      target="blank"
    >
      <div
        class="cursor-pointer relative flex flex-col justify-content-center \\   items-center "
      >
        <div
          class="mt-4 w-full rounded-md overflow-hidden"
        >
          <img
            alt="test"
            class="h-[16rem] sm:h-[20rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-2"
        >
          <p
            class="text-lightg text-md"
          >
            Vegan Fajitas 9
          </p>
        </div>
        <img
          class="absolute top-5 right-1 hover:scale-125 hover:cursor-pointer"
          data-testid="pin"
          height="30px"
          src="/thumb-tack-cancel.webp"
          width="30px"
        />
      </div>
    </a>
    <a
      href="https://www.feastingathome.com/vegan-fajitas/"
      target="blank"
    >
      <div
        class="cursor-pointer relative flex flex-col justify-content-center \\   items-center "
      >
        <div
          class="mt-4 w-full rounded-md overflow-hidden"
        >
          <img
            alt="test"
            class="h-[16rem] sm:h-[20rem] w-full object-cover"
            src="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
          />
        </div>
        <div
          class="m-2"
        >
          <p
            class="text-lightg text-md"
          >
            Vegan Fajitas 10
          </p>
        </div>
        <img
          class="absolute top-5 right-1 hover:scale-125 hover:cursor-pointer"
          data-testid="pin"
          height="30px"
          src="/thumb-tack-cancel.webp"
          width="30px"
        />
      </div>
    </a>
  </div>
</div>
`;
